build/nlu-regression.js

```diff
@@ -0,0 +1,218 @@
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(data)
+  })
+  const login = JSON.parse(rawLogin)
+
+  const token = login.payload && login.payload.token
+  return login.statusCode ? undefined : token
+}
+
+async function createBot(botId, token) {
+  const newBot = JSON.stringify({
+    id: botId,
+    name: 'testy',
+    template: {
+      id: 'bp-nlu-regression-testing',
+      moduleId: 'nlu-testing'
+    },
+    category: undefined
+  })
+
+  await post('/api/v1/admin/bots', newBot, {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json',
+    'Content-Length': Buffer.byteLength(newBot),
+    'X-BP-Workspace': 'default'
+  })
+}
+
+async function waitForTraining(botId, token) {
+  return new Promise(function(resolve) {
+    var i = 0
+    console.log(`training...`)
+    const intervalId = setInterval(async () => {
+      const raw = await get(`/api/v1/bots/${botId}/mod/nlu/train`, {
+        Authorization: `Bearer ${token}`
+      })
+      const trainingStatus = JSON.parse(raw)
+      if (!trainingStatus.isTraining) {
+        clearInterval(intervalId)
+        resolve()
+      } else {
+        console.log(`training... ${2 * ++i}s`)
+      }
+    }, 2000)
+  })
+}
+
+function round(n, acc) {
+  var num = Math.pow(10, acc)
+  return Math.round(n * num) / num
+}
+
+async function runAllTests(botId, token) {
+  const baseNluTesting = `/api/v1/bots/${botId}/mod/nlu-testing`
+  const rawTests = await get(`${baseNluTesting}/tests`, {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json'
+  })
+  const allTests = JSON.parse(rawTests)
+  const nTests = allTests.length
+  let nPassing = 0
+
+  var i = 0
+  for (const test of allTests) {
+    const retry = () =>
+      post(`${baseNluTesting}/tests/${test.id}/run`, '', {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json'
+      })
+
+    let rawResult
+    try {
+      rawResult = await retry()
+    } catch (err) {
+      console.error(err, 'retrying')
+      rawResult = await retry()
+    }
+
+    const testResult = JSON.parse(rawResult)
+    nPassing += testResult.success ? 1 : 0
+    console.log(`(${i++} /${nTests}) #${test.id}`, 'success: ', testResult.success)
+  }
+
+  const acc = (nPassing / nTests) * 100
+  return round(acc, 1)
+}
+
+async function compareScore(score) {
+  const latestResultsFile = `${nluTestingDir}/src/bot-templates/bp-nlu-regression-testing/latest-results.csv`
+  const latestResultsContent = fs.readFileSync(latestResultsFile, { encoding: 'utf8' })
+  const previousScoreOccurence = latestResultsContent.match(/summary: ((100|\d{1,2})[.]\d{1})?/gm)
+  if (!previousScoreOccurence || !previousScoreOccurence[0]) {
+    return false
+  }
+
+  const previousScoreString = previousScoreOccurence[0].split(':')[1]
+  const previousScore = parseFloat(previousScoreString)
+  console.log('previous score was: ', previousScore)
```

Comment:
Since we're displaying this in the console, can you capitalize your messages so it displays nicely?